File: src/main/java/org/jdc/template/model/webservice/websearch/WebSearchService.java
package org.jdc.template.model.webservice.websearch;

import org.jdc.template.model.webservice.websearch.dto.DtoSearchResponse;

import okhttp3.ResponseBody;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;
import retrofit2.http.Streaming;
import retrofit2.http.Url;

public interface WebSearchService {
    String BASE_URL = "https://ajax.googleapis.com";

    @GET("/ajax/services/search/web?v=1.0")
    Call<DtoSearchResponse> search(@Query("q") String query);

    @GET
    Call<DtoSearchResponse> searchByFullUrl(@Url String url);

    @Streaming
    @GET("/ajax/services/search/web?v=1.0")
    Call<ResponseBody> searchToFile(@Query("q") String query);
}

File: src/main/kotlin/org/jdc/template/model/webservice/websearch/WebSearchService.kt

package org.jdc.template.model.webservice.websearch

import okhttp3.ResponseBody
import org.jdc.template.model.webservice.websearch.dto.DtoSearchResponse
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query
import retrofit2.http.Streaming
import retrofit2.http.Url

interface WebSearchService {

    @GET("/ajax/services/search/web?v=1.0")
    fun search(@Query("q") query: String): Call<DtoSearchResponse>

    @GET
    fun searchByFullUrl(@Url url: String): Call<DtoSearchResponse>

    @Streaming
    @GET("/ajax/services/search/web?v=1.0")
    fun searchToFile(@Query("q") query: String): Call<ResponseBody>

    companion object {
        val BASE_URL = "https://ajax.googleapis.com"
    }
}
